unit_tests/UnitTestLinearSystem.h

```diff
@@ -198,7 +198,11 @@ class TestLinearSystem : public sierra::nalu::LinearSystem
 
   sierra::nalu::CoeffApplier* get_coeff_applier()
   {
-    return new TestCoeffApplier(lhs_, rhs_, numSumIntoCalls_, isEdge_, numDof_);
+    if (hostCoeffApplier.get() == nullptr) {
```

Comment:
With `std::unique_ptr`, you can just do 

```C++
if (!hostCoeffApplier) {
  // initialization code here
}
```

Reply:
See https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool

Reply:
ok, I'm testing that change (it occurred in a few other places in my previous commit). I'll push a commit for that soon.